change-viewpoint/src/main/java/com/esri/arcgisruntime/sample/changeviewpoint/MainActivity.java

```diff
@@ -0,0 +1,160 @@
+        mCenterScaleButton = (Button) findViewById(R.id.centerScaleButton);
+        mAnimateButton = (Button) findViewById(R.id.animateButton);
+
+        // create geometry of Griffith Park from JSON raw file, add graphics and set viewpoint of the map view to Griffith Park
+        mGeometryButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // create an input stream for the raw text file containing JSON of Griffith Park
+                InputStream ins = getResources().openRawResource(
+                        getResources().getIdentifier("griffithparkjson",
+                                "raw", getPackageName()));
+
+                InputStreamReader inputReader = new InputStreamReader(ins);
+
+                BufferedReader bufferReader = new BufferedReader(inputReader);
+                String line;
+                StringBuilder text = new StringBuilder();
+
+                // read the text file
+                try {
+                    while (( line = bufferReader.readLine()) != null) {
+                        text.append(line);
+                    }
+                } catch (IOException e) {
+                    Log.d(TAG, e.toString());
+                }
+                String JsonString = text.toString();
+
+                // create Geometry from JSON
+                Geometry geometry = Geometry.fromJson(JsonString, spatialReference);
+                GraphicsOverlay overlay = new GraphicsOverlay();
+                // add graphics overlay on map view
+                mMapView.getGraphicsOverlays().add(overlay);
+                SimpleFillSymbol fillSymbol = new SimpleFillSymbol(SimpleFillSymbol.Style.DIAGONAL_CROSS, Color.GREEN, null);
+                // add graphic of Griffith Park
+                overlay.getGraphics().add(new Graphic(geometry, fillSymbol));
+
+                // set viewpoint of map view to Geometry - Griffith Park
+                mMapView.setViewpointGeometryAsync(geometry);
+
+                mGeometryButton.setBackgroundColor(getResources().getColor(R.color.primary_dark));
+                mAnimateButton.setBackgroundColor(getResources().getColor(R.color.primary));
+                mCenterScaleButton.setBackgroundColor(getResources().getColor(R.color.primary));
+            }
+        });
+
+        mAnimateButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // create the London location point
+                Point londonPoint = new Point(28677947.756181,22987445.6186465, spatialReference);
+                // create the viewpoint with the London point and scale
+                Viewpoint viewpoint = new Viewpoint(londonPoint, SCALE);
+                // set the map views's viewpoint to London with a ten second duration
+                mMapView.setViewpointWithDurationAsync(viewpoint, 10);
```

Comment:
Since the view points were scaled to 5000, increasing the animation time doesn't have any effects on transition as map is displaced before it can be loaded. In order to do so, I have set the duration as is but instead changed the SCALE while transitioning from Griffith Park to London. The new scaling factor is square of SCALE.